File: src/woodpuzzle/solver/DFSSolver.java
package woodpuzzle.solver;

import woodpuzzle.model.Configuration;
import woodpuzzle.model.Puzzle;

public class DFSSolver extends AbstractSolver {
	public DFSSolver(Puzzle puzzle) {
		super(puzzle);
	}

	@Override
	public Configuration findSolution() {
		DFSStrategy traversal = new DFSStrategy(this.getPuzzle());
		
		this.generateRootConfigs(new Configuration(this.getPuzzle()));
		
		for (Configuration c : rootConfigs) {
			try {
				traversal.traverse(new ConfigurationTreeNode(null, c));
			} catch (FoundException e) {
				return e.config;
			} catch (EndException e) {
				System.out.println("Unexpected exception in HaltingDFSSolver: ");
				e.printStackTrace();
			}
		}
		
		return null;
	}
}

File: src/woodpuzzle/solver/DFSSolver.kt
package woodpuzzle.solver

import woodpuzzle.model.Configuration
import woodpuzzle.model.Puzzle

class DFSSolver(override val puzzle: Puzzle) : Solver {
    override fun findSolution(): Configuration? {
        val traversal = DFSTraversal(puzzle)
        val rootConfig = Configuration(puzzle)
        val rootNode = ConfigurationTreeNode(null, rootConfig)
        try {
            traversal.traverse(rootNode)
        } catch (e: FoundException) {
            return e.config
        } catch (e: EndException) {
            println("Unexpected exception in DFSSolver: ")
            e.printStackTrace()
        }
        return null
    }
}
